packages/react-storage/src/components/StorageBrowser/Views/Controls/AddFiles.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
+import type { OmitElements } from '../types';
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+import { useControl } from '../../context/controls';
+
+const { Button: ButtonElement } = StorageBrowserElements;
+
+const BLOCK_NAME = `${CLASS_BASE}__add-files`;
+
+export interface _AddFilesControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> {
+  (): React.JSX.Element;
+  Button: T['Button'];
+}
+
+export interface AddFilesControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends OmitElements<_AddFilesControl<T>, 'Button'> {
+  (): React.JSX.Element;
+}
+
+const Button = withBaseElementProps(ButtonElement, {
+  className: `${BLOCK_NAME}`,
+  variant: 'primary',
+});
```

Comment:
We don't need to create a wrapper for every `element`, the `withBaseElementProps` util was meant to simplify scenarios where only a className or another static prop was required by a component that had no functional usage. As this `Button` is used directly in `AddFilesControl`, the `className` and `variant` can just be passed as `props` in the JSX